<?xml version="1.0" encoding="UTF-8" ?>
<rss version="2.0" xmlns:atom="http://www.w3.org/2005/Atom" xmlns:dc="http://purl.org/dc/elements/1.1/">
  <channel>
    <title>Why getting error while installing ansible pack on stackstorm version &quot;st2 3.3.0, on Python 2.7.5&quot;</title>
    <link>https://forum.stackstorm.com/t/why-getting-error-while-installing-ansible-pack-on-stackstorm-version-st2-3-3-0-on-python-2-7-5/1605</link>
    <description>Hi Guys,

I have installed stackstorm 3.3.0 version on Python 2.7.5 and i want to install Ansible Pack but while installing i am getting below error--

    Traceback (most recent call last):
    File &quot;/opt/stackstorm/st2/lib/python2.7/site-packages/python_runner/python_action_wrapper.py&quot;, line 334, in &lt;module&gt;
    obj.run()
    File &quot;/opt/stackstorm/st2/lib/python2.7/site-packages/python_runner/python_action_wrapper.py&quot;, line 193, in run
    output = action.run(**self._parameters)
    File &quot;/opt/stackstorm/packs/packs/actions/pack_mgmt/download.py&quot;, line 86, in run
    logger=self.logger)
    File &quot;/opt/stackstorm/st2/lib/python2.7/site-packages/st2common/util/pack_management.py&quot;, line 165, in download_pack
    verify_pack_version(pack_metadata=pack_metadata, use_python3=use_python3)
    File &quot;/opt/stackstorm/st2/lib/python2.7/site-packages/st2common/util/pack_management.py&quot;, line 464, in verify_pack_version
    raise ValueError(msg)
    ValueError: Pack &quot;ansible&quot; requires Python 3.x, but current Python version is &quot;2.7.5&quot;. You can override this restriction by providing the &quot;force&quot; flag, but the pack is not guaranteed to work.

after the googling even i tried to install with --force flag but installation is getting timeout here is output..

    [root@stackstorm-2 packs]# st2 pack install ansible --force
    For the &quot;ansible&quot; pack, the following content will be registered:
    rules     |  0
    sensors   |  0
    triggers  |  0
    actions   |  8
    aliases   |  0
    Installation may take a while for packs with many items.
            [ succeeded ] init_task
            [ succeeded ] download_pack
            [ succeeded ] make_a_prerun
            [ succeeded ] get_pack_dependencies
            [ succeeded ] check_dependency_and_conflict_list
            [  timeout  ] install_pack_requirements
    id: 6054f442cc9b23f2f2c47796
    action.ref: packs.install
    parameters:
      force: true
      packs:
      - ansible
      python3: false
    status: failed
    start_timestamp: Fri, 19 Mar 2021 18:58:10 UTC
    end_timestamp: Fri, 19 Mar 2021 19:09:33 UTC
    result:
      errors: []
      output:
        conflict_list: []
        message: &#39;&#39;
        packs_list:
        - ansible
        warning_list: null
    +--------------------------+-------------------------+------------------------------+-----------------------------+-------------------------------+
    | id                       | status                  | task                         | action                      | start_timestamp               |
    +--------------------------+-------------------------+------------------------------+-----------------------------+-------------------------------+
    | 6054f443b8d4a06354e7e968 | succeeded (0s elapsed)  | init_task                    | core.noop                   | Fri, 19 Mar 2021 18:58:11 UTC |
    | 6054f443b8d4a06354e7e978 | succeeded (57s elapsed) | download_pack                | packs.download              | Fri, 19 Mar 2021 18:58:11 UTC |
    | 6054f47db8d4a06354e7e988 | succeeded (13s elapsed) | make_a_prerun                | packs.virtualenv_prerun     | Fri, 19 Mar 2021 18:59:09 UTC |
    | 6054f48bb8d4a06354e7e998 | succeeded (7s elapsed)  | get_pack_dependencies        | packs.get_pack_dependencies | Fri, 19 Mar 2021 18:59:23 UTC |
    | 6054f493b8d4a06354e7e9a8 | succeeded (0s elapsed)  | check_dependency_and_conflic | core.noop                   | Fri, 19 Mar 2021 18:59:31 UTC |
    |                          |                         | t_list                       |                             |                               |
    | 6054f494b8d4a06354e7e9b8 | timeout (600s elapsed)  | install_pack_requirements    | packs.setup_virtualenv      | Fri, 19 Mar 2021 18:59:32 UTC |
    +--------------------------+-------------------------+------------------------------+-----------------------------+-------------------------------+
    [root@stackstorm-2 packs]#
cam someone please help me with pointers to fix this issue?


Thanks</description>
    
    <lastBuildDate>Mon, 10 May 2021 17:16:15 +0000</lastBuildDate>
    <category>Pack Management</category>
    <atom:link href="https://forum.stackstorm.com/t/why-getting-error-while-installing-ansible-pack-on-stackstorm-version-st2-3-3-0-on-python-2-7-5/1605.rss" rel="self" type="application/rss+xml" />
      <item>
        <title>Why getting error while installing ansible pack on stackstorm version &quot;st2 3.3.0, on Python 2.7.5&quot;</title>
        <dc:creator><![CDATA[VladoPortos]]></dc:creator>
        <description><![CDATA[
            <p>No update ? I have the same issue on RH7 with StackStorm installed directly in OS. ( I have instances where StackStorm was deployed in docker and there its working fine )</p>
<p>Dependencies are met:</p>
<pre><code>Package gcc-4.8.5-44.el7.x86_64 already installed and latest version
Package krb5-devel-1.15.1-50.el7.x86_64 already installed and latest version
</code></pre>
<p>Solved for me it was not enough space in /tmp, the installation easily eats 1.5GB of space in /tmp</p>
          <p><a href="https://forum.stackstorm.com/t/why-getting-error-while-installing-ansible-pack-on-stackstorm-version-st2-3-3-0-on-python-2-7-5/1605/6">Read full topic</a></p>
        ]]></description>
        <link>https://forum.stackstorm.com/t/why-getting-error-while-installing-ansible-pack-on-stackstorm-version-st2-3-3-0-on-python-2-7-5/1605/6</link>
        <pubDate>Mon, 10 May 2021 16:41:35 +0000</pubDate>
        <guid isPermaLink="false">forum.stackstorm.com-post-1605-6</guid>
        <source url="https://forum.stackstorm.com/t/why-getting-error-while-installing-ansible-pack-on-stackstorm-version-st2-3-3-0-on-python-2-7-5/1605.rss">Why getting error while installing ansible pack on stackstorm version &quot;st2 3.3.0, on Python 2.7.5&quot;</source>
      </item>
      <item>
        <title>Why getting error while installing ansible pack on stackstorm version &quot;st2 3.3.0, on Python 2.7.5&quot;</title>
        <dc:creator><![CDATA[bharat19th]]></dc:creator>
        <description><![CDATA[
            <p>Hi Guys,</p>
<p>does anyone has the idea why it is failing?</p>
<p>Thanks</p>
          <p><a href="https://forum.stackstorm.com/t/why-getting-error-while-installing-ansible-pack-on-stackstorm-version-st2-3-3-0-on-python-2-7-5/1605/5">Read full topic</a></p>
        ]]></description>
        <link>https://forum.stackstorm.com/t/why-getting-error-while-installing-ansible-pack-on-stackstorm-version-st2-3-3-0-on-python-2-7-5/1605/5</link>
        <pubDate>Mon, 22 Mar 2021 08:28:14 +0000</pubDate>
        <guid isPermaLink="false">forum.stackstorm.com-post-1605-5</guid>
        <source url="https://forum.stackstorm.com/t/why-getting-error-while-installing-ansible-pack-on-stackstorm-version-st2-3-3-0-on-python-2-7-5/1605.rss">Why getting error while installing ansible pack on stackstorm version &quot;st2 3.3.0, on Python 2.7.5&quot;</source>
      </item>
      <item>
        <title>Why getting error while installing ansible pack on stackstorm version &quot;st2 3.3.0, on Python 2.7.5&quot;</title>
        <dc:creator><![CDATA[bharat19th]]></dc:creator>
        <description><![CDATA[
            <p>Hi as per your suggestion i tried installing ansible version 0.5.9 but it got time out</p>
<pre><code>[root@stackstorm-2 ~]# st2 pack install ansible=0.5.9

        [ succeeded ] init_task
        [ succeeded ] download_pack
        [ succeeded ] make_a_prerun
        [ succeeded ] get_pack_dependencies
        [ succeeded ] check_dependency_and_conflict_list
        [  timeout  ] install_pack_requirements

id: 60564b79a2834016ce11486f
action.ref: packs.install
parameters:
  packs:
  - ansible=0.5.9
  python3: false
status: failed
start_timestamp: Sat, 20 Mar 2021 19:22:33 UTC
end_timestamp: Sat, 20 Mar 2021 19:33:25 UTC
result:
  errors: []
  output:
    conflict_list: []
    message: ''
    packs_list:
    - ansible
    warning_list: null
+--------------------------+-------------------------+------------------------------+-----------------------------+-------------------------------+
| id                       | status                  | task                         | action                      | start_timestamp               |
+--------------------------+-------------------------+------------------------------+-----------------------------+-------------------------------+
| 60564b7a7b79aa316fbc62dc | succeeded (0s elapsed)  | init_task                    | core.noop                   | Sat, 20 Mar 2021 19:22:34 UTC |
| 60564b7b7b79aa316fbc62ec | succeeded (44s elapsed) | download_pack                | packs.download              | Sat, 20 Mar 2021 19:22:35 UTC |
| 60564ba77b79aa316fbc62fc | succeeded (1s elapsed)  | make_a_prerun                | packs.virtualenv_prerun     | Sat, 20 Mar 2021 19:23:19 UTC |
| 60564ba97b79aa316fbc630c | succeeded (1s elapsed)  | get_pack_dependencies        | packs.get_pack_dependencies | Sat, 20 Mar 2021 19:23:21 UTC |
| 60564bab7b79aa316fbc631c | succeeded (0s elapsed)  | check_dependency_and_conflic | core.noop                   | Sat, 20 Mar 2021 19:23:23 UTC |
|                          |                         | t_list                       |                             |                               |
| 60564bab7b79aa316fbc632c | timeout (601s elapsed)  | install_pack_requirements    | packs.setup_virtualenv      | Sat, 20 Mar 2021 19:23:23 UTC |
+--------------------------+-------------------------+------------------------------+-----------------------------+-------------------------------+
[root@stackstorm-2 ~]#</code></pre>
          <p><a href="https://forum.stackstorm.com/t/why-getting-error-while-installing-ansible-pack-on-stackstorm-version-st2-3-3-0-on-python-2-7-5/1605/4">Read full topic</a></p>
        ]]></description>
        <link>https://forum.stackstorm.com/t/why-getting-error-while-installing-ansible-pack-on-stackstorm-version-st2-3-3-0-on-python-2-7-5/1605/4</link>
        <pubDate>Sat, 20 Mar 2021 19:51:50 +0000</pubDate>
        <guid isPermaLink="false">forum.stackstorm.com-post-1605-4</guid>
        <source url="https://forum.stackstorm.com/t/why-getting-error-while-installing-ansible-pack-on-stackstorm-version-st2-3-3-0-on-python-2-7-5/1605.rss">Why getting error while installing ansible pack on stackstorm version &quot;st2 3.3.0, on Python 2.7.5&quot;</source>
      </item>
      <item>
        <title>Why getting error while installing ansible pack on stackstorm version &quot;st2 3.3.0, on Python 2.7.5&quot;</title>
        <dc:creator><![CDATA[bharat19th]]></dc:creator>
        <description><![CDATA[
            <p>Yes Pre-req are in place-- below are the output-</p>
<pre><code> [root@stackstorm-2 ~]# yum install gcc krb5-devel
    Loaded plugins: fastestmirror
    Loading mirror speeds from cached hostfile
     * atomic: mirrors.hosting.in.th
     * base: mirrors.piconets.webwerks.in
     * epel: ftp.jaist.ac.jp
     * extras: centos.excellmedia.net
     * updates: centos.excellmedia.net
    Package gcc-4.8.5-44.el7.x86_64 already installed and latest version
    Package krb5-devel-1.15.1-50.el7.x86_64 already installed and latest version
    Nothing to do
    [root@stackstorm-2 ~]#
</code></pre>
<p>Thanks</p>
          <p><a href="https://forum.stackstorm.com/t/why-getting-error-while-installing-ansible-pack-on-stackstorm-version-st2-3-3-0-on-python-2-7-5/1605/3">Read full topic</a></p>
        ]]></description>
        <link>https://forum.stackstorm.com/t/why-getting-error-while-installing-ansible-pack-on-stackstorm-version-st2-3-3-0-on-python-2-7-5/1605/3</link>
        <pubDate>Sat, 20 Mar 2021 19:06:58 +0000</pubDate>
        <guid isPermaLink="false">forum.stackstorm.com-post-1605-3</guid>
        <source url="https://forum.stackstorm.com/t/why-getting-error-while-installing-ansible-pack-on-stackstorm-version-st2-3-3-0-on-python-2-7-5/1605.rss">Why getting error while installing ansible pack on stackstorm version &quot;st2 3.3.0, on Python 2.7.5&quot;</source>
      </item>
      <item>
        <title>Why getting error while installing ansible pack on stackstorm version &quot;st2 3.3.0, on Python 2.7.5&quot;</title>
        <dc:creator><![CDATA[amanda11]]></dc:creator>
        <description><![CDATA[
            <p>Have you installed the pre-reqs mentioned on the ansible pack for gcc etc: <a href="https://github.com/StackStorm-Exchange/stackstorm-ansible#requirements" class="inline-onebox" rel="noopener nofollow ugc">GitHub - StackStorm-Exchange/stackstorm-ansible: st2 content pack containing ansible integrations</a>?</p>
<p>The v1.0.0 is only verified on python3, but I’m not sure anythign was added to make it not work with python2. But alternatively you could tell it to install v0.5.9 which is the last version tagged for python2 support (you can specify a particular verison of a pack to install)</p>
          <p><a href="https://forum.stackstorm.com/t/why-getting-error-while-installing-ansible-pack-on-stackstorm-version-st2-3-3-0-on-python-2-7-5/1605/2">Read full topic</a></p>
        ]]></description>
        <link>https://forum.stackstorm.com/t/why-getting-error-while-installing-ansible-pack-on-stackstorm-version-st2-3-3-0-on-python-2-7-5/1605/2</link>
        <pubDate>Sat, 20 Mar 2021 18:41:41 +0000</pubDate>
        <guid isPermaLink="false">forum.stackstorm.com-post-1605-2</guid>
        <source url="https://forum.stackstorm.com/t/why-getting-error-while-installing-ansible-pack-on-stackstorm-version-st2-3-3-0-on-python-2-7-5/1605.rss">Why getting error while installing ansible pack on stackstorm version &quot;st2 3.3.0, on Python 2.7.5&quot;</source>
      </item>
      <item>
        <title>Why getting error while installing ansible pack on stackstorm version &quot;st2 3.3.0, on Python 2.7.5&quot;</title>
        <dc:creator><![CDATA[bharat19th]]></dc:creator>
        <description><![CDATA[
            <p>Hi Guys,</p>
<p>I have installed stackstorm 3.3.0 version on Python 2.7.5 and i want to install Ansible Pack but while installing i am getting below error–</p>
<pre><code>Traceback (most recent call last):
File "/opt/stackstorm/st2/lib/python2.7/site-packages/python_runner/python_action_wrapper.py", line 334, in &lt;module&gt;
obj.run()
File "/opt/stackstorm/st2/lib/python2.7/site-packages/python_runner/python_action_wrapper.py", line 193, in run
output = action.run(**self._parameters)
File "/opt/stackstorm/packs/packs/actions/pack_mgmt/download.py", line 86, in run
logger=self.logger)
File "/opt/stackstorm/st2/lib/python2.7/site-packages/st2common/util/pack_management.py", line 165, in download_pack
verify_pack_version(pack_metadata=pack_metadata, use_python3=use_python3)
File "/opt/stackstorm/st2/lib/python2.7/site-packages/st2common/util/pack_management.py", line 464, in verify_pack_version
raise ValueError(msg)
ValueError: Pack "ansible" requires Python 3.x, but current Python version is "2.7.5". You can override this restriction by providing the "force" flag, but the pack is not guaranteed to work.
</code></pre>
<p>after the googling even i tried to install with --force flag but installation is getting timeout here is output…</p>
<pre><code>[root@stackstorm-2 packs]# st2 pack install ansible --force
For the "ansible" pack, the following content will be registered:
rules     |  0
sensors   |  0
triggers  |  0
actions   |  8
aliases   |  0
Installation may take a while for packs with many items.
        [ succeeded ] init_task
        [ succeeded ] download_pack
        [ succeeded ] make_a_prerun
        [ succeeded ] get_pack_dependencies
        [ succeeded ] check_dependency_and_conflict_list
        [  timeout  ] install_pack_requirements
id: 6054f442cc9b23f2f2c47796
action.ref: packs.install
parameters:
  force: true
  packs:
  - ansible
  python3: false
status: failed
start_timestamp: Fri, 19 Mar 2021 18:58:10 UTC
end_timestamp: Fri, 19 Mar 2021 19:09:33 UTC
result:
  errors: []
  output:
    conflict_list: []
    message: ''
    packs_list:
    - ansible
    warning_list: null
+--------------------------+-------------------------+------------------------------+-----------------------------+-------------------------------+
| id                       | status                  | task                         | action                      | start_timestamp               |
+--------------------------+-------------------------+------------------------------+-----------------------------+-------------------------------+
| 6054f443b8d4a06354e7e968 | succeeded (0s elapsed)  | init_task                    | core.noop                   | Fri, 19 Mar 2021 18:58:11 UTC |
| 6054f443b8d4a06354e7e978 | succeeded (57s elapsed) | download_pack                | packs.download              | Fri, 19 Mar 2021 18:58:11 UTC |
| 6054f47db8d4a06354e7e988 | succeeded (13s elapsed) | make_a_prerun                | packs.virtualenv_prerun     | Fri, 19 Mar 2021 18:59:09 UTC |
| 6054f48bb8d4a06354e7e998 | succeeded (7s elapsed)  | get_pack_dependencies        | packs.get_pack_dependencies | Fri, 19 Mar 2021 18:59:23 UTC |
| 6054f493b8d4a06354e7e9a8 | succeeded (0s elapsed)  | check_dependency_and_conflic | core.noop                   | Fri, 19 Mar 2021 18:59:31 UTC |
|                          |                         | t_list                       |                             |                               |
| 6054f494b8d4a06354e7e9b8 | timeout (600s elapsed)  | install_pack_requirements    | packs.setup_virtualenv      | Fri, 19 Mar 2021 18:59:32 UTC |
+--------------------------+-------------------------+------------------------------+-----------------------------+-------------------------------+
[root@stackstorm-2 packs]#
</code></pre>
<p>cam someone please help me with pointers to fix this issue?</p>
<p>Thanks</p>
          <p><a href="https://forum.stackstorm.com/t/why-getting-error-while-installing-ansible-pack-on-stackstorm-version-st2-3-3-0-on-python-2-7-5/1605/1">Read full topic</a></p>
        ]]></description>
        <link>https://forum.stackstorm.com/t/why-getting-error-while-installing-ansible-pack-on-stackstorm-version-st2-3-3-0-on-python-2-7-5/1605/1</link>
        <pubDate>Sat, 20 Mar 2021 16:50:12 +0000</pubDate>
        <guid isPermaLink="false">forum.stackstorm.com-post-1605-1</guid>
        <source url="https://forum.stackstorm.com/t/why-getting-error-while-installing-ansible-pack-on-stackstorm-version-st2-3-3-0-on-python-2-7-5/1605.rss">Why getting error while installing ansible pack on stackstorm version &quot;st2 3.3.0, on Python 2.7.5&quot;</source>
      </item>
  </channel>
</rss>
